### ISSUE TITLE
Block storage Web API (random-access-storage compliant)

### ISSUE THREAD

**Issue by pfrazee** [beakerbrowser/beaker#1222]
This has been a common request from people building database tools (eg @substack and @mafintosh).

The API should conform to [random-access-storage](https://github.com/random-access-storage/random-access-storage). I think we can scope the data access to an origin (similar to localStorage) and provide multiple openable namespaces within that. The total data usage would be metered (maybe 100MB free with more space requestable?).

Pulling from random-access-storage we might do this API:

```js
var storage = blockStorage(name)
storage.opened // bool
storage.closed // bool
storage.destroyed // bool
storage.addEventListener('open')
storage.addEventListener('close')
storage.addEventListener('destroy')
storage.open(cb)
storage.read(offset, size, callback)
storage.write(offset, buffer, [callback])
storage.del(offset, size, [callback])
storage.stat(callback)
storage.close([callback])
storage.destroy([callback])
```

We can't be 100% equivalent to `random-access-storage` because the event systems differ between node and the Web. I also removed the "readable/writable/deletable/statable" boolean flags but I would be fine having those in there (they'd all be true).

**Comment by karissa** [comment]
Yes, this would be nice so that we could easily upgrade/modify existing codebase running on electron to be beaker-compatible. There a few apps I have in mind on my `to-build` list coming up early next year for this..
